include/mgba/internal/gba/hardware-extensions.h

```diff
@@ -0,0 +1,56 @@
+/* Copyright (c) 2021 Jeffrey Pfau
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+#ifndef HARDWARE_EXTENSIONS_H
+#define HARDWARE_EXTENSIONS_H
+
+#include <mgba-util/common.h>
+
+#include <mgba/core/log.h>
+#include <mgba/core/timing.h>
+
+#include <mgba/internal/gba/io.h>
+#include <mgba/internal/gba/hardware-extensions-ids.h>
+
+
+#define REG_HWEX_VERSION_VALUE HWEX_EXTENSIONS_COUNT
+#define HWEX_MORE_RAM_SIZE 0x100000 // 1 MB
```

Comment:
Why is this specifically 1 MB? It seems arbitrary. Perhaps making it adjustable would be better.

Reply:
No particular reason, it was easier to code at the moment. I added three more commands to the extension:
- Init: to set the set the initial size of the extra RAM.
- Resize: to change it. It will either truncate the data at the end or add more spece to the end.
- Destroy: frees the extra RAM.
Also added a max size clause of 64 MB. Don't think any gba game would need more than that, but it can be changed.